kornia/geometry/solvers/polynomial_solver.py

```diff
@@ -0,0 +1,181 @@
+    r"""Solve given quadratic equation.
+
+    The function takes the coefficients of quadratic equation and returns the solutions.
+
+    Args:
+        coeffs : The coefficients of quadratic equation :`(B, 3)`
+
+    Returns:
+        A tensor of shape `(B, 2)` containing the solutions to the quadratic equation.
+    """
+
+    KORNIA_CHECK_SHAPE(coeffs, ['B', '3'])
+
+    # Coefficients of quadratic equation
+    a = coeffs[:, 0]  # coefficient of x^2
+    b = coeffs[:, 1]  # coefficient of x
+    c = coeffs[:, 2]  # constant term
+
+    # Calculate discriminant
+    delta = b * b - 4 * a * c
+
+    # Create masks for negative and zero discriminant
+    mask_negative = delta < 0
+    mask_zero = delta == 0
+
+    # Calculate 1/(2*a) for efficient computation
+    inv_2a = 0.5 / a
+
+    # Initialize solution tensor
+    solution = torch.zeros((coeffs.shape[0], 2), device=coeffs.device, dtype=coeffs.dtype)
+
+    # Handle cases with zero discriminant
+    if torch.any(mask_zero):
+        solution[mask_zero, 0] = -b[mask_zero] * inv_2a[mask_zero]
+        solution[mask_zero, 1] = solution[mask_zero, 0]
+
+    # Handle cases with negative discriminant
+    # if torch.any(mask_negative):
+    #     solution[mask_negative, :] = torch.tensor(0, device=coeffs.device, dtype=coeffs.dtype)
+
+    sqrt_delta = torch.sqrt(delta)
+
+    # Handle cases with non-negative discriminant
+    mask = torch.bitwise_and(~mask_negative, ~mask_zero)
+    if torch.any(mask):
+        solution[mask, 0] = (-b[mask] + sqrt_delta[mask]) * inv_2a[mask]
+        solution[mask, 1] = (-b[mask] - sqrt_delta[mask]) * inv_2a[mask]
+
+    return solution
+
+
+def solve_cubic(coeffs: Tensor) -> Tensor:
+    r"""Solve given cubic equation.
+
+    The function takes the coefficients of cubic equation and returns
+    the solutions.
+
+    Args:
+        coeffs : The coefficients cubic equation : `(B, 4)`
+
+    Returns:
+        A tensor of shape `(B, 3)` containing the solutions to the cubic equation.
+    """
+    KORNIA_CHECK_SHAPE(coeffs, ['B', '4'])
+
+    _PI = torch.tensor(math.pi, device=coeffs.device, dtype=coeffs.dtype)
+
+    # Coefficients of cubic equation
+    a = coeffs[:, 0]  # coefficient of x^3
+    b = coeffs[:, 1]  # coefficient of x^2
+    c = coeffs[:, 2]  # coefficient of x
+    d = coeffs[:, 3]  # constant term
+
+    solutions = torch.zeros((len(coeffs), 3), device=a.device, dtype=a.dtype)
+
+    mask_a_zero = a == 0
+    mask_b_zero = b == 0
+    mask_c_zero = c == 0
+
+    mask_a_zero & mask_b_zero & mask_c_zero
+    mask_first_order = mask_a_zero & mask_b_zero & ~mask_c_zero
+    mask_second_order = mask_a_zero & ~mask_b_zero & ~mask_c_zero
+
+    if torch.any(mask_second_order):
+        solutions[mask_second_order, 0:2] = solve_quadratic(coeffs[mask_second_order, 1:])
+
+    if torch.any(mask_first_order):
+        solutions[mask_first_order, 0] = torch.tensor(1.0, device=a.device, dtype=a.dtype)
+
+    # if torch.any(mask_zero_order):
```

Comment:
same here

Reply:
In both cases, the algorithm needs to branch out to handle those scenarios. However, since the 'solution' is already initialized with zero, it won't make any changes to the result.
I kept those commented codes for algorithm readability

Reply:
Then add that explanation in short as a comment